source/Api.cpp

```diff
@@ -59,6 +69,10 @@ namespace Aws
 
         ApiHandle::~ApiHandle()
         {
+            ReleaseStaticDefaultClientBootstrap();
+            ReleaseStaticDefaultEventLoopGroup();
+            ReleaseStaticDefaultHostResolver();
```

Comment:
Add a test to ensure that, when these default objects get automatically created, their memory gets cleaned up. Could probably be as simple as a test that just creates the ClientBootstrap (resulting in the other 2 getting created)